drivers/mil_passive_sonar/scripts/multilateration

```diff
@@ -57,25 +67,82 @@ class MultilaterationNode(object):
         except tf2_ros.TransformException, e:
             rospy.logwarn('TF Exception: {}'.format(e))
             return
+
+        # Check if two samples were taken too close to each other
+        if self.last_pos is not None and np.linalg.norm(self.last_pos - origin) < self.minimum_delta_dist:
+            rospy.loginfo("Observation too close to previous, ignoring")
+            return
+        self.last_pos = origin
+
+        # Add ping to samples list
         self.vec_samples.append((origin, origin + vec))
-        if len(self.vec_samples) == self.vec_samples.maxlen:
+
+        # If there are enough samples, calculate position
+        if len(self.vec_samples) >= self.buffer_min:
             self.publish_position()
 
     def publish_position(self):
-        line_array = np.array([(p[0][0], p[0][1], p[1][0], p[1][1]) for p in self.vec_samples])
+        # Setup lines from samples
+        line_array = np.array([(np.array(p[0][0:2]), np.array(p[1][0:2])) for p in self.vec_samples])
+
+        # Calculate least squares intersection
         where = self.ls_line_intersection2d(line_array)
+
+        # If no point could be determined, don't publish
+        if where is None:
+            return
```

Comment:
This is a nice place to do a `rospy.logwarn_throttle(10.0, "could not ...")`. So if nothing gets published the logs show why

Reply:
Done